src/main/java/org/codehaus/plexus/archiver/AbstractUnArchiver.java

```diff
@@ -316,9 +317,17 @@ public void setIgnorePermissions( final boolean ignorePermissions )
         this.ignorePermissions = ignorePermissions;
     }
 
-    protected void extractFile( final File srcF, final File dir, final InputStream compressedInputStream,
-                                String entryName, final Date entryDate, final boolean isDirectory,
-                                final Integer mode, String symlinkDestination, final FileMapper[] fileMappers )
+    protected File extractFile(final File srcF, final File dir, final InputStream compressedInputStream,
+                               String entryName, final Date entryDate, final boolean isDirectory,
+                               final Integer mode, String symlinkDestination, final FileMapper[] fileMappers)
+            throws IOException, ArchiverException {
+        return extractFile(srcF, dir, compressedInputStream, Long.MAX_VALUE, entryName, entryDate, isDirectory,
+                mode, symlinkDestination, fileMappers);
+    }
+
+    protected File extractFile(final File srcF, final File dir, final InputStream compressedInputStream,
+                               Long remainingSpace, String entryName, final Date entryDate, final boolean isDirectory,
```

Comment:
Also would it make sense to have different name for this method (as the other `extractFile` does not return `File`) to avoid confusion. I'm not saying it is. I'm just wondering.